src/main/java/com/google/starfish/servlets/FavoriteNoteServlet.java

```diff
@@ -0,0 +1,133 @@
+    res.setContentType("application/json");
+    if(!validateUser(req)) {
+      res.getWriter().println(false);
+      return;
+    }
+    HttpSession activeSession = req.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+    Long noteId = Long.valueOf(req.getParameter("note_id"));
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+    try(Connection conn = pool.getConnection()) {
+      String stmt =
+          "SELECT EXISTS "
+        + "(SELECT * "
+        + "FROM favorite_notes "
+        + "WHERE user_id=? "
+        + "AND note_id=?);";
+
+      try (PreparedStatement favStmt = conn.prepareStatement(stmt)) {
+        favStmt.setString(1, userId);
+        favStmt.setLong(2, noteId);
+        ResultSet rs = favStmt.executeQuery();
+        rs.next();
+        res.getWriter().println(rs.getBoolean(1));
+      }
+    } catch (SQLException ex) {
+      System.err.print(ex);
+    }
+  }
+
+  @Override
+  public void doPost(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    if(!validateUser(req)) {
+      res.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    HttpSession activeSession = req.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+    Long noteId = Long.valueOf(req.getParameter("note_id"));
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");  
+
+    try (Connection conn = pool.getConnection()) {
+      favoriteNoteService.insertFavoriteNote(pool, noteId, userId);
+      String authorId = noteService.getNoteByNoteId(pool, noteId).getAuthorId();
+      userService.increasePointsOnFavorite(pool, authorId);
+    } catch (SQLException ex) {
+      System.err.print(ex);
+    }
+  }
+
+  @Override
+  public void doDelete(HttpServletRequest req, HttpServletResponse res) throws ServletException, IOException {
+    if(!validateUser(req)) {
+      res.setStatus(HttpServletResponse.SC_FORBIDDEN);
+      return;
+    }
+
+    HttpSession activeSession = req.getSession(false);
+    String userId = (String) activeSession.getAttribute("user_id");
+    Long noteId = Long.valueOf(req.getParameter("note_id"));
+    DataSource pool = (DataSource) req.getServletContext().getAttribute("my-pool");
+
+    try (Connection conn = pool.getConnection()) {
+      favoriteNoteService.deleteRowByCompoundId(pool, noteId, userId);
+      String authorId = noteService.getNoteByNoteId(pool, noteId).getAuthorId();
+      userService.decreasePointsOnUnfavorite(pool, authorId);
+    } catch (SQLException ex) {
+      System.err.print(ex);
+    }
+  }
+
+  /**
+   * Validates the user using the request's session and cookies. If there is a
+   * valid user logged in, returns true. Otherwise, returns false.
+   **/
+  private boolean validateUser(HttpServletRequest req) {
```

Comment:
If this logic is spread across each servlet, could you consolidate the logic and share it through a utility service?

Reply:
Yep, I'll do that!